portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,47 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Seun</h1>");
-  }
+    private ArrayList<String> messages;
+
+    @Override
+    public void init(){
+        messages = new ArrayList<>();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+        String json = convertToJsonUsingGson(messages);
+
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    private String convertToJsonUsingGson(ArrayList arr) {
+        Gson gson = new Gson();
+        String json = gson.toJson(arr);
+        return json;
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+        String input = getParameter(request, "text-input", "");
+
+        messages.add(input);    // adds to array
+
+        response.sendRedirect("/index.html"); //take me home
+    }
+
+    /**
+     * @return the request parameter, or the default value if the parameter
+     *         was not specified by the client
+    */
```

Comment:
The asterisks should be aligned vertically, and the second line of the comment should line up with the top line on the left (there shouldn't be extra white space between the asterisk and that text)